integrations/freshchat/src/client.ts

```diff
@@ -0,0 +1,99 @@
+import { RuntimeError } from '@botpress/client'
+import axios, { Axios } from 'axios'
+import { FreshchatAgent, FreshchatChannel, FreshchatConfiguration, FreshchatUser } from './definitions/schemas'
+import * as bp from '.botpress'
+
+// API docs: https://developers.freshchat.com/api/
+
+class FreshchatClient {
+  private _client: Axios
+
+  public constructor(private _config: FreshchatConfiguration, private _logger: bp.Logger) {
+    this._client = axios.create({
+      baseURL: `https://${this._config.domain}.freshchat.com/v2`,
+    })
+
+    this._client.interceptors.request.use((axionsConfig) => {
```

Comment:
**question:** why use an interceptor for simply passing the `Authorization` header ? why not passing at the ctor ?

Reply:
This code is repeated several times during the client during the http calls, so by using the client with the interceptors I make sure to always have the correct header + clean code

Reply:
I know but you can set the header at line 13 when constructing the axios instance

Reply:
Got your point, the real advantage of using interceptors in this case would be to use a function inside of it, which is not being done today, that would allow the client to always use the last token without changing the client, even if the value is changed without a get/set function, I will revisit this logic in newer versions